### ISSUE TITLE
Listing All Devices

### ISSUE THREAD

**Issue by halil-raso** [NetHome/ZWave#4]
Hi @stefangs 
Is there any way to list learned devices and print them?

Thanks in advance.

**Comment by stefangs** [comment]
Yes, to get a list of all nodes in the network you use the command: GetInitData
That will return among other things a list of the node identities of all added nodes. After that you can use the RequestNodeInfo command to get more information about a specific node. There are more commands that you may use to find out additional info about the node depending on the capabilities of the node. If you look in the ZWaveNode-class in the NetHomeServer, you will see a state machine that does discovery of a ZWave node. The process is like this:

        Discovery state machine:

         * Routing info (GetRoutingInfoMessageClass())
         * is the node failed? (IsFailedNodeMessageClass)
         * Request node info (RequestNodeInfoMessageClass)
         * If node supports Manufacturer specific command class, request it to get manufacturer info (manufacturerSpecific.getManufacturerSpecificMessage())
         * If node supports version command class, loop through all command classes and request version
         * Check version of interface (versionCommandClass.getVersionMessage())
         * If node supports MultiInstance, get all endpoints? (multiInstance.initEndpoints(stageAdvanced))

**Comment by csantiestebantoca** [comment]
Hi @stefangs 
I can't understand very well jet how this library works… please could you explain me better how to list the connected devices.
Thanks
